### PR TITLE
add analysis directory

### PR DESCRIPTION
**Pull request by bintiibrahim**
user can define path to directory in `config.yaml`, ~~must not have any spaces~~ can have spaces!
`analysisDirectory: "C:/users/rina/Workflow1"`

analysisDirectory will contain:
- user input fastq files if any 
- snakemake output

ensembl output is in `data` and SnpEff output is in its own `SnpEff` folder

In a nutshell: before executing workflow, user must
- define `sra` or `fq1`,`fq2` in config.yaml
- if defined, clear `analysisDirectory` or define new directory
 else clear `data`
- put `.fastq` files in `analysisDirectory` folder if any
- clear `SnpEff/data`

TO DO
- [x] default analysisDirectory `output` if not defined by user
- [x] have `data/uniprot` and `data/ensembl` for reuse
- [ ] rule for cleaning SnpEff?

### CODE REVIEW

File: rules/align.smk [hunk 1]
```diff
@@ -69,13 +69,13 @@ rule hisat2_align_bam:
 
 rule hisat2_merge_bams:
     input:
-        bams=expand("data/{sra}.sorted.bam", sra=config["sra"]) if check_sra() is True else expand("data/{fq1}.sorted.bam", fq1=config["fq1"])
+        bams=expand("{{dir}}/{sra}.sorted.bam", sra=config["sra"]) if check_sra() is True else expand("{{dir}}/{fq1}.sorted.bam", fq1=config["fq1"])
```
Comment (acesnik) [inline rules/align.smk]: What do the double brackets do?

**Review by acesnik — COMMENTED** [review]
Looks good.

File: rules/align.smk (hunk 1 above)
Comment (bintiibrahim) [reply rules/align.smk]: it prevents from expanding the `dir` wildcard while expanding the `sra` wildcard

Comment (bintiibrahim) [reply rules/align.smk]: without the double brackets, `dir` undefined error would be thrown because it's attempting to expand `dir`

Comment (acesnik) [reply rules/align.smk]: Cool!

**Pull request merged**